### PR TITLE
Expanded seats so a character with some gear and a weapon can enter

### PR DESCRIPTION
**Pull request by VictoriaStudios**
#### Summary
Balance "Expand seats to enter with some armor on"

#### Purpose of change

Lately, the new way of calculating the volume of an entering player character on a sat tile led to a frustrating situation: Although only wearing some medium-tier armor and a weapon (sometimes just the armor!), the player couldn't enter the tile. Since driving around and scavenging is the main part of the mid-game, this led to frustrating situations where backpacks/armor/weapons were left behind due to game enforcing seat capacity limits that have not been updated in at least 6 months.

#### Describe the solution

With the exception of the 0L seat, all seat capacity was increased by 20L as to not make this buff ridiculous. This allows a normal human in ballistic armor to take a seat in a Humvee - as was actually done in e.g. Iraq.
EDIT: Also added 20 L to aisles for consistency.

#### Describe alternatives you've considered

Alternatively, the calculations of player volume could be rolled back, however, this seems an easier and more practical solution.

#### Testing

Wearing different armors and weapons, I entered the vehicle from multiple angles.

#### Additional context

CDDA always needs to  balance realism with playability. The inability to easily enter and exit vehicles in the mid-game with some armor on enforces relatively arbitrary seat volume number while heavily inhibiting playability.

### CODE REVIEW

**Comment by fairyarmadillo** [comment]
Chiming in to report that when I initially set the vehicle seat sizes, I did so rather arbitrarily. Since a seat is not an enclosed container, it's difficult to say exactly how much could fit there without toppling over into other tiles or whatever. 

I picked the old values based on the idea that a seat can fit a little more than a human body. Since human bodies became a bit bigger and started tracking gear following the recent body volume changes, the best solution is to make the seats a bit bigger too. This PR accomplishes that without making them so big that it defeats the purpose, so IMO this is the way to go.

**Comment by VictoriaStudios** [comment]
> Chiming in to report that when I initially set the vehicle seat sizes, I did so rather arbitrarily. Since a seat is not an enclosed container, it's difficult to say exactly how much could fit there without toppling over into other tiles or whatever.
> 
> I picked the old values based on the idea that a seat can fit a little more than a human body. Since human bodies became a bit bigger and started tracking gear following the recent body volume changes, the best solution is to make the seats a bit bigger too. This PR accomplishes that without making them so big that it defeats the purpose, so IMO this is the way to go.

Many thanks for your input!

**Comment by misterprimus** [comment]
Just started a new game as a morbidly obese backpacker. Started in a car and could not enter the driver's seat. Dropped backpack and cleared off the front seat completely, no go. Something tells me it's not working properly as is...

**Comment by kevingranade** [comment]
There is most definitely a point where people can't fit in a car, the question is where that point is and what mitigations are present for it when it happens.
At first look, morbidly obese is exactly the demographic that will not fit. Need further details for your character to decide if that seems reasonable or not

**Comment by misterprimus** [comment]
> There is most definitely a point where people can't fit in a car, the question is where that point is and what mitigations are present for it when it happens. At first look, morbidly obese is exactly the demographic that will not fit. Need further details for your character to decide if that seems reasonable or not

I guess it comes down to how obese "morbidly obese" is supposed to be. The My 600-lb Life contestants certainly don't all fit into the front seat (either one) and must travel in the back seat. However, this level of obesity generally leaves people barely able to stand and walk for limited amounts of time. The morbidly obese CDDA character can outrun/outsprint some zeds, so either XXXL characters are too fast for realism or too big for it. They should either be adjusted to fit into the front seat of the car or slowed down to not be able to outrun zeds.

**Comment by kevingranade** [comment]
Clinically, morbidly obese starts at a BMI of 40. At average height (5'6") that's 250lbs, which isn't having much trouble fitting into a car,  and at 6' it's around 300lbs, so the question is, is that right on the cusp,  or well over? This is why I asked for more information.

Yes obese characters should have large movement penalties,  but that's a seperate issue.

**Comment by misterprimus** [comment]
> Clinically, morbidly obese starts at a BMI of 40. At average height (5'6") that's 250lbs, which isn't having much trouble fitting into a car, and at 6' it's around 300lbs, so the question is, is that right on the cusp, or well over? This is why I asked for more information.
> 
> Yes obese characters should have large movement penalties, but that's a seperate issue.

Well in my case it's this:
image

**Comment by PatrikLundell** [comment]
Please debug spawn a prison bus and do the test with it. In my game my 175 cm character (overweight, only reaching normal at the end of that same day) couldn't get into the driver's seat of such a vehicle even when completely naked, which is obviously bonkers, since I'm fairly sure the penal system doesn't hire dwarfs explicitly for the purpose of them being able to drive their buses (and they may not be able to either, due to vision and ability to reach pedal issues).

@kevingranade: The game refuses to provide the player with the information you request. You get the height from the image, but the weight is unknown, as the bathroom scale tops out at a paltry 150 kg. There may be some debug info way to get at the info, though, but I don't know what it is.

**Comment by VictoriaStudios** [comment]
> Please debug spawn a prison bus and do the test with it. In my game my 175 cm character (overweight, only reaching normal at the end of that same day) couldn't get into the driver's seat of such a vehicle even when completely naked, which is obviously bonkers, since I'm fairly sure the penal system doesn't hire dwarfs explicitly for the purpose of them being able to drive their buses (and they may not be able to either, due to vision and ability to reach pedal issues).
> 
> @kevingranade: The game refuses to provide the player with the information you request. You get the height from the image, but the weight is unknown, as the bathroom scale tops out at a paltry 150 kg. There may be some debug info way to get at the info, though, but I don't know what it is.

There you go, spawned with my PR, 175, overweight, wearing ballistic vest and wielding M4.

Screenshot_20240705_081903
Screenshot_20240705_081917

**Comment by MaraStel** [comment]
Can you also apply a similar increase to aisles so they are at least consistent with the largest seats again?

**Comment by VictoriaStudios** [comment]
> Can you also apply a similar increase to aisles so they are at least consistent with the largest seats again?

Done! Many thanks for the hint.

**Comment by misterprimus** [comment]
Shooting from the hip here, but does it make sense to reduce the value for purposes of calculating whether the character fits somewhere (or perhaps bypass it entirely) or not if the character is in prone stance to simulate squeezing through/moving things around to get through an area at the cost of time?

**Comment by zachary-kaelan** [comment]
> I picked the old values based on the idea that a seat can fit a little more than a human body.

Someone can definitely sit on someone else's lap in a car seat. A hefty backpack can also be held in one's lap.

**Comment by fairyarmadillo** [comment]
> > I picked the old values based on the idea that a seat can fit a little more than a human body.
> 
> Someone can definitely sit on someone else's lap in a car seat. A hefty backpack can also be held in one's lap.

At the time, worn and held items did not count whatsoever.

**Comment by kevingranade** [comment]
> > If you don't want your seat belt to work sure.

> That sounds like a good tradeoff to me (in the game, not the real world).

It's a problem because it creates a situation where seat belts are no longer automatic, get in the car with your backpack on your lap, ignore the prompt about the seatbelt, hit a tree, die. So if you do want to use your seatbelt, you have to intuitively stash your backpack (and potentially other things) elsewhere, or take some manual action to use it, or set an option somewhere to insist on using the seatbelt over the "convinience" of not using it.  Alternately the non-safe option of holding your backpack in your lap is hidden behind a default-off option, and we're carrying around support for this nonsensical course of action for the handful of people that want to self-destructively prioritize holding stuff in their lap over being protected from being flung out of a vehicle.
Also driving with a backpack in your lap is pretty meaningfully impaired, are we adding that too?

Nothing about this fixes anything or makes anything better, it requires extra infrastructure to support a thing it makes no sense to do in the first place.

Alternately, just add a QOL feature where when you get in the car it pops up a menu that allows you to take off your bulky items and stash them somewhere in reach (passenger seat, back seat) instead of just telling you you can't fit, then when you exit the vehicle you automatically grab them and re-wear them.  For bonus points this can remember your decision and auto-stash them. Done, no longer any meaningful friction anywhere other than actually being too bulky to fit in certain vehicles, which is intended.

**Comment by PatrikLundell** [comment]
Unfortunately, the current implementation doesn't allow you to stash your items in the space specifically intended for stashing stuff, namely the storage space between the seats in e.g. flatbeds (that, additionally, suffer from having the smaller seats) because you can't get into the seat with your stuff, and thus can't reach the storage space.
I had to take off my worn stuff (a holy golf bag of long stuff storage and a backpack) at the bed of the truck in order to be able to enter without leaving the carrying capacity completely behind. Neither of the containers was anywhere near full, but just containing stuff I actually want to keep at hand.

**Comment by misterprimus** [comment]
> Unfortunately, the current implementation doesn't allow you to stash your items in the space specifically intended for stashing stuff, namely the storage space between the seats in e.g. flatbeds (that, additionally, suffer from having the smaller seats) because you can't get into the seat with your stuff, and thus can't reach the storage space. I had to take off my worn stuff (a holy golf bag of long stuff storage and a backpack) at the bed of the truck in order to be able to enter without leaving the carrying capacity completely behind. Neither of the containers was anywhere near full, but just containing stuff I actually want to keep at hand.

In the most recent version I tested I couldn't even get into the empty front seat WITHOUT a backpack/weapon just due to being morbidly obese, which officially doesn't require one to be freakishly huge. So this won't really fix the problem completely.

**Comment by molkemon** [comment]
> Clinically, morbidly obese starts at a BMI of 40. At average height (5'6") that's 250lbs, which isn't having much trouble fitting into a car, and at 6' it's around 300lbs, so the question is, is that right on the cusp, or well over? This is why I asked for more information.
> 
> Yes obese characters should have large movement penalties, but that's a seperate issue.

Well yes but also no, the current implementation of morbidily obese is 500k kcal at start. My character is 170cm and barely below 500k kcal. That is only 65kg of pure fat. With skeltal and the rest this would amount to at worst like 160-170kg total. He cannot fit into any vehicle whatsover and that is nonsense. The way the current systems of fat vs volume vs space in cars work, doesn't work properly at all.

**Comment by kevingranade** [comment]
> That is only 65kg of pure fat.

Yea sure 70L of fat no big deal hardly overweight at all. That's only about the volume of an average person, so this character is literally 2x larger than the average person due to the fat they're carrying around, but I'm sure getting in a car is no trouble at all.

The hyperbole isn't helping. What we need are:
"ok how much room is there in various cars anyway"
  Headroom? Hiproom? Does footroom matter? (probably not)
"How do we represent the variance between vehicles without making everything suck"
"how do we handle that with constructed vehicles"
"how do we map from total height/weight or other existing stats to effective volume?"
"can we pull that off without adding more stats, because more stats sucks"

What isn't helping at all is people just doing drive-by comments where all they do is complain vaguely that their morbidly obese character can't fit in a car.  I'm not the slightest bit convinced that your literally double sized character *should* be able to fit in most cars, but I haven't run the numbers in any detail.  Should there be vehicles they do fit in? sure thing, see my above points about what needs to happen, but this takes time and we're not going to get done any faster based on volume of complaints.

If this is intolerable for you, roll back to an older experimental or stable from before these changes happened, that's the deal with experimental.

**Comment by zachary-kaelan** [comment]
Personally, I already have stow boards next to my vehicle doors just to quickly dump encumbering stuff into before a fight, without having to board the vehicle. Modifying vehicles to accommodate one's playstyle is a huge part of the game. If someone starts the game morbidly obese or extremely tall then I say tough shit, most cars should be off limits to them until they figure out how to create their own accommodations. Specialized vehicle parts might need to be added, but that's easy enough.

**Comment by PatrikLundell** [comment]
If you think realism at the expense of tedium is good, then yes, you can start to restrict what characters can bring into the driver's seat of their vehicles (I don't think that adds to the game, apart from the mutant monstrosities that are clearly described not to fit, so players should already know that will be an issue).

Now, if you do insist on it, whatever you come up with should fulfill some basic criteria:
- It should be predictable, i.e. players should be able to quickly learn what the rules are.
- It should be reasonable. Punishing players for randomizing their characters' height and rolling a tall one is bad. Normal length people at reasonably normal weights shouldn't be blocked from using vehicles. I would probably start causing trouble (but not a blanket ban) when heading into morbidly obese, but not before that. Extremely tall people can hit the limit earlier, as that's an explicit player choice. I'd rather use morbidly obese as the criterion than an arbitrary volume, because it's a visible criterion the game readily displays (although that should be checked against NPCs->companions: one of the beggars is morbidly obese, I believe, and blocking the player from bringing all of the beggars back to the base would be very bad game design).
- It should work with the information actually present in the game (possibly by increasing the info in it to allow that). This means that the real world observation that morbidly obese people probably won't fit in extreme sports cars (at least not reversibly) but seem to typically use SUVs to move around should only be applied to the game if the game is able to distinguish between these kinds of vehicles (including player modified/constructed ones) and convey the difference to the players. Currently the game doesn't provide any distinctions, so the least restricting case should be used for the model.
- It shouldn't trap characters in situations they can't get out of (apart from smashing up the vehicle). I've had several cases where I managed to get into the middle back seat from the driver's seat, only to be unable to move back or in any other direction. I've been lucky enough to have been able to drop my worn containers into available seats/storages so far, but one day they'll all be full of loot (especially since moving around inside is more common when space is filling up, so you try to fill space further away).
- Note that juggling your own equipment is tedious and annoying, but having to do so to the standard companion Liam, whose "huge bulk" complains about entering vehicles every time is even worse (I haven't had to do so yet, but I also haven't used him as a mule yet).

**Comment by zachary-kaelan** [comment]
> My character is 170cm and barely below 500k kcal. That is only 65kg of pure fat. With skeltal and the rest this would amount to at worst like 160-170kg total

That's a BMI of 59. 40 or higher is morbidly obese. There wasn't even room for 59 on the chart I'm looking at.

**Comment by fairyarmadillo** [comment]
> That's a BMI of 59. 40 or higher is morbidly obese. There wasn't even room for 59 on the chart I'm looking at.

Is XXXL making people too big? AFAIK it just flatly sets a huge amount of stored calories, and may be going beyond what is reasonable.

**Comment by misterprimus** [comment]
> > That is only 65kg of pure fat.
> 
> Yea sure 70L of fat no big deal hardly overweight at all. That's only about the volume of an average person, so this character is literally 2x larger than the average person due to the fat they're carrying around, but I'm sure getting in a car is no trouble at all.
> 
> The hyperbole isn't helping. What we need are: "ok how much room is there in various cars anyway" Headroom? Hiproom? Does footroom matter? (probably not) "How do we represent the variance between vehicles without making everything suck" "how do we handle that with constructed vehicles" "how do we map from total height/weight or other existing stats to effective volume?" "can we pull that off without adding more stats, because more stats sucks"
> 
> What isn't helping at all is people just doing drive-by comments where all they do is complain vaguely that their morbidly obese character can't fit in a car. I'm not the slightest bit convinced that your literally double sized character _should_ be able to fit in most cars, but I haven't run the numbers in any detail. Should there be vehicles they do fit in? sure thing, see my above points about what needs to happen, but this takes time and we're not going to get done any faster based on volume of complaints.
> 
> If this is intolerable for you, roll back to an older experimental or stable from before these changes happened, that's the deal with experimental.

As little as 300 lbs is already morbidly obese for most people.
image

https://my.clevelandclinic.org/health/diseases/21989-class-iii-obesity-formerly-known-as-morbid-obesity

300 lbs isn't exactly crazy huge.
image

We're not talking _need-to-rent-a-cargo-van-to-take-him-to-the-hospital_ obese here.

**Comment by kevingranade** [comment]
> We're not talking need-to-rent-a-cargo-van-to-take-him-to-the-hospital obese here.

Yes we are actually,  you haven't specified how big your character actually is so I don't have any idea how big they are.  As such all you are contributing here is noise,  stop.

**Comment by misterprimus** [comment]
> > We're not talking need-to-rent-a-cargo-van-to-take-him-to-the-hospital obese here.
> 
> Yes we are actually, you haven't specified how big your character actually is so I don't have any idea how big they are. As such all you are contributing here is noise, stop.

What do you mean by how big? I wasn't aware the game had a numerical pounds/KG parameter for characters.

**Comment by PatrikLundell** [comment]
The game calculates a mass based on height and stored calories. The height is readily available, but the mass is harder to get at. The game contains a bathroom scale item you can use to measure your character's weight, but it tops out at 150 kg, so it's useless for the heavyweight characters who would be most likely to have use for it.

**Comment by misterprimus** [comment]
> The game calculates a mass based on height and stored calories. The height is readily available, but the mass is harder to get at. The game contains a bathroom scale item you can use to measure your character's weight, but it tops out at 150 kg, so it's useless for the heavyweight characters who would be most likely to have use for it.

So it's a debug-only stat? What's the procedure to pull it from the game?

**Comment by VictoriaStudios** [comment]
So um .. when are the decisions on PRs usually made? Again, I maintain that the change is reasonable, minimal and won't break any other mechanics I can think of or the autotests have found... No idea why the compilation builds fail, but it's not due to slightly changed JSON values.
Btw, the discussions on heavily modifying seat entering mechanics really are unproductive... This PR won't change it and, seeing as its a marginal topic in terms of balance but has the potential to greatly decrease gaming joy, I say a decision on the currently proposed PR should be taken.

**Comment by misterprimus** [comment]
> Well yes but also no, the current implementation of morbidily obese is 500k kcal at start. My character is 170cm and barely below 500k kcal. That is only 65kg of pure fat. With skeltal and the rest this would amount to at worst like 160-170kg total. He cannot fit into any vehicle whatsover and that is nonsense. The way the current systems of fat vs volume vs space in cars work, doesn't work properly at all.

It indeed does seem like the seat restrictions are bit more restrictive than real life would have one believe. Here is an extremely obese person who still somehow manages to fit in AND operate a motor vehicle: https://www.youtube.com/watch?v=eD4R01AZujU

**Comment by RenechCDDA** [comment]
> So um .. when are the decisions on PRs usually made? Again, I maintain that the change is reasonable, minimal and won't break any other mechanics I can think of or the autotests have found... No idea why the compilation builds fail, but it's not due to slightly changed JSON values. Btw, the discussions on heavily modifying seat entering mechanics really are unproductive... This PR won't change it and, seeing as its a marginal topic in terms of balance but has the potential to greatly decrease gaming joy, I say a decision on the currently proposed PR should be taken.

This is not a full answer to you, but you'll definitely need to lint your changes before this can be merged, willing or not. Merge permissions are specifically disabled if it fails the json style check (which your changes have).

The bot automatically posted suggestions for them starting [here](https://github.com/CleverRaven/Cataclysm-DDA/pull/74897#pullrequestreview-2161841767). You can also use the json formatter binary which ships with the game, or the online linter mentioned in the docs. Alternatively let me know if you have any issues and I can push them from here for you.

**Comment by npostavs** [comment]
> > The game contains a bathroom scale item you can use to measure your character's weight, but it tops out at 150 kg, so it's useless for the heavyweight characters who would be most likely to have use for it.
> 
> So it's a debug-only stat? What's the procedure to pull it from the game?

I don't see anything directly relevant in the debug menu, but you can spawn a livestock scale instead of a bathroom scale. https://cdda-guide.nornagon.net/item/livestock_scale

**Comment by RenechCDDA** [comment]
Friends, please do not post unless you have something constructive to say about the contents of this PR or something constructive to say about an issue that someone else has raised. I understand that this is currently a topic of great interest to many of you, but tangential comments are *not* helping to get this PR merged or resolve any issues. I have marked several of the latest as off-topic to hide them by default.

**Comment by misterprimus** [comment]
> > We're not talking need-to-rent-a-cargo-van-to-take-him-to-the-hospital obese here.
> 
> Yes we are actually, you haven't specified how big your character actually is so I don't have any idea how big they are. As such all you are contributing here is noise, stop.

Debug kcal report and other screens:
image
image

Inventory at time of not fitting into empty front seat:
image

For kcal calculation:
image

Calculation:
image

But that's just the stored fat. We need to figure out the weight at which a 5'7" (or thereabout) human would die from starvation and add that "dry weight" of the human, so to speak, to arrive at the total weight.

If we take an average height person:
https://www.britannica.com/biography/Christopher-McCandless

`
McCandless would have returned to civilization in early July 1992 had he been able to recross the Teklanika, but the river he had easily forded two months earlier was by then swollen with snowmelt, and he returned to the bus. Forced to continue on a substandard diet, he weakened and died in mid-August. His body, which weighed only 67 pounds (30.4 kg), was discovered by hunters on September 6. The cause of death was officially reported as starvation.
`

In other words, my fictional C:DDA character should weigh 30 kg "dry" weight + 72 kg stored fat = 102 kg. Let's round up since I rounded down before and it's 103 kg, or 227 lbs:
image

Hope this helps!

**Comment by fairyarmadillo** [comment]
I don't think it fully counters your argument, but McCandless was found in a state of decomposition that would have removed a lot of weight from his body. I assume he might have been 5 or 10 kg heavier when he passed.

**Comment by misterprimus** [comment]
> I don't think it fully counters your argument, but McCandless was found in a state of decomposition that would have removed a lot of weight from his body. I assume he might have been 5 or 10 kg heavier when he passed.

I expected that, so I looked up some concentration camp stories, and got close figures. For example:
https://www.bbc.com/news/world-europe-51204090
`At this point, Henri Kichka, a tall young man of 19, weighed 39kg (85lb) and to this day he suffers from the injuries he sustained from the long march on broken and bleeding feet through the snows of January in Eastern Europe.`

And this man is described as "tall." So he'd have more "dry weight." Granted, for that time and in a different country.
image
(He also still had enough self-consumable body mass to still be alive)

Even then we're talking less than 250 lbs total if we add back the fat.

**Comment by Maddremor** [comment]
The core issue here is that cargo volume and seating/aisle volume is not equivalent. Continually bumping up the storage capacity of seating until stuff works is not the solution.

**Comment by fairyarmadillo** [comment]
> The core issue here is that cargo volume and seating/aisle volume is not equivalent. Continually bumping up the storage capacity of seating until stuff works is not the solution.

The capacity for a human is going to almost always be smaller than the theoretical amount of inanimate objects you could fit in a place, since a person is a single, rigid, oddly-shaped object that needs to be able to move and breathe. Making a seat big enough to hold a human is not going to cause anything strange to happen with the amount of cargo that can fit there.

Furthermore, the existing seat sizes are ones I arbitrarily chose to make the seats fit characters under the assumption that a medium human is 75 liters and a large one is 92, with no volume changes from gear, height, or BMI. That assumption is no longer valid because people have gotten bigger even without gear, but those numbers are now for some reason being treated as sacrosanct, which is the source of the current trouble.

**Comment by kevingranade** [comment]
> The capacity for a human is going to almost always be smaller than the theoretical amount of inanimate objects you could fit in a place, since a person is a single, rigid, oddly-shaped object that needs to be able to move and breathe. Making a seat big enough to hold a human is not going to cause anything strange to happen with the amount of cargo that can fit there.

This is false.  A pile of items has an angle of repose that limits how much stuff you can put on it.  If you try to put more items on such a pile, it doesn't just stay there, it falls off the pile.  Meanwhile a person can occupy that space without falling off of it by holding themselves in position. Yes in a vehicle you can keep adding items and use the bounds of the car to trap them in place, but *this is not a functional way of storing things in a vehicle*, it's going to have all kinds of side effects like you being unable to use the vehicle, items shifting around and falling out of the vehicle, etc. As such the volume a vehicle part can contain isn't the total free volume, it's a measure of how much stuff you can put in that part before it stops functioning correctly  For a driver's seat in particular, if you pile enough stuff on it that the floorboards get covered and you can no longer operate the pedals the driver's seat has been rendered inoperable. For any other seat the limit is higher, but not by much more, if you fill up the footwell you can't sit normally in the seat anymore, and if the level of stuff keeps climbing from there you quickly reach a point where you have an unstable pile that will collapse, especially once the vehicle is in motion. For an aisle there is a loose metric of when the floor gets covered and there is no longer walking space, once this is exceeded you have to start stepping on items, which we don't represent in the game.

Meanwhile your number for "can a person fit in this space" isn't directly related to volume, it's more about the specific dimensions of the space and which dimension is the limiting one.  In e.g. a driver's seat, the limiting dimensions are hiproom, headroom, and seatback-to-steering wheel.  In an aisle it's mostly just hiproom. We don't necessarily need to model those numbers explicitly, but any solution to this issue needs to recognize that those are the constraints we are dealing with, not pure volume.

> those numbers are now for some reason being treated as sacrosanct, which is the source of the current trouble.

This is a mischaracterization of what is happening.  The holdup of *this PR in particular* is simply that we don't have workable examples of who should and should not be able to fit, and discussion of what those examples should be is ongoing. "just shuffle the numbers around until people stop complaining" is not going to get us to a stable solution, so we need a rationale for who should and should not be able to fit, and then tune the numbers such that those examples act properly.
The fundamental problem is not related to cargo capacity values at all, it's that cargo capacity is not the right number to use for "can a particular creature occupy this space".

> But that's just the stored fat. We need to figure out the weight at which a 5'7" (or thereabout) human would die from starvation and add that "dry weight" of the human, so to speak, to arrive at the total weight.

This is not a reasonable way to caclulate this, people who are starving and especially who have died of starvation have also lost a significant amount of lean body mass and water, so this massively under reports total body weight.  What you're looking for is calculating lean body mass vs fat mass (using i.e. https://www.calculator.net/lean-body-mass-calculator.html) and solving for ~558,000kcal of fat. That solution is around 307lbs/139kg from what I can gather, though there's a decent amount of variance based on different estimates.  I had expected it to be pretty straightforward to get from there to hip diameter and therefore "can this person fit in an average car", but I'm drawing a blank searching for a formula to predict that relation.  https://bmcmedresmethodol.biomedcentral.com/articles/10.1186/1471-2288-12-115 claims to do that but isn't loading, WBM has it but I don't have time to unpack it right now https://web.archive.org/web/20240625131315/https://bmcmedresmethodol.biomedcentral.com/articles/10.1186/1471-2288-12-115

**Comment by PatrikLundell** [comment]
Apart from aisles getting covered by stuff that would get damaged and/or slow movement when trodden on, they're also extremely poor storage spaces since there's little to keep stuff in place, so it shuffles around as the vehicle moves (and ships, in particular, can sink due to cargo shifting when it's incorrectly/not secured in poor weather, although that's not modeled in the game).
I think the old model of aisles not providing any storage while seats provide a minimal amount is better than trying to use storage capacity for both storage capacity and seating capacity. While it's true you can place a fair amount of stuff in a seat if there's nobody there, that amount of stuff has to be suitably shaped. A cardboard box of a suitable size can fit in a seat (and possibly not move around dangerously while driven), but the stuff in it can not be placed safely on top of the seat without that support. It would be a fair bit of work to try to model such dependencies in a somewhat realistic way that's also possible to convey to the player (and that's assuming someone is actually willing to implement and tune it).

**Comment by zachary-kaelan** [comment]
From https://bmcmedresmethodol.biomedcentral.com/articles/10.1186/1471-2288-12-115:
* For men: `22.61306 + 2.520738 * BMI + 0.1583812 * AGE - 0.6570163 * (RACE == BLACK) + 0.1818819 * (ETHNICITY == HISPANIC)`
* For women: `28.81919 + 2.218007 * BMI − 3.688953 * (AGE >= 35) + 0.125975 * AGE * (AGE >= 35) − 0.6570163 * (RACE == BLACK) + 0.1818819 * (ETHNICITY == HISPANIC)`

Bone structure and fat distribution on the body varies by gender and genetics, which can complicate things.

**Comment by misterprimus** [comment]
> This is not a reasonable way to caclulate this, people who are starving and especially who have died of starvation have also lost a significant amount of lean body mass and water, so this massively under reports total body weight. What you're looking for is calculating lean body mass vs fat mass (using i.e. https://www.calculator.net/lean-body-mass-calculator.html) and solving for ~558,000kcal of fat. That solution is around 307lbs/139kg from what I can gather, though there's a decent amount of variance based on different estimates. I had expected it to be pretty straightforward to get from there to hip diameter and therefore "can this person fit in an average car", but I'm drawing a blank searching for a formula to predict that relation. https://bmcmedresmethodol.biomedcentral.com/articles/10.1186/1471-2288-12-115 claims to do that but isn't loading, WBM has it but I don't have time to unpack it right now https://web.archive.org/web/20240625131315/https://bmcmedresmethodol.biomedcentral.com/articles/10.1186/1471-2288-12-115 turns out this paper isn't good either as it ignores BMI in excess of 40, which is exactly what we're interested in.

Wouldn't it make more sense to simply look up someone with BMI in excess of 40 and see if they can fit into a car?
https://www.youtube.com/watch?v=SX4A8QTvUYI

image
image
image

**Comment by kevingranade** [comment]
I did get the formula eventually, but I don't know how useful it is as it's a curve fit to strictly BMI < 40 samples, which is not what we're interested in.
Hoping it's *somewhat* predictive, we can run mrprimus' character though and see what we get...
22.61306 + 2.520738 * 48 + 0.1583812 * 30 = 147.34 cm = 18.5" hip width.  Average and typical stats for cars seem to be a bit hard to come by, but glancing through several popular SUVs I see hiproom around 28" (often reported in total hiproom across both seats so in the 56" range), and even compact cars like the honda civic only drop to about 27".

Reversing the formula above, we're looking for a waist circumference of about 215cm, so that implies a BMI around 56.5.  Now that's hip room, with a number that large it leaves me wondering if seatback-to-steering wheel becomes the dominant factor, IDK.

Another outcome of it being that high is the observation that that's not a playable character, so hip width might not ever be a limiting factor, potentially leaving us with height vs legroom/headroom as the only meaningful limiter.

**Comment by fairyarmadillo** [comment]
> I think the old model of aisles not providing any storage while seats provide a minimal amount is better than trying to use storage capacity for both storage capacity and seating capacity.

This would re-introducd old issues such as not being able to store things in seats which should reasonably fit there and oversized/overloaded 
characters being able to fit places they ought not to.

Splitting passenger capacity out to its own value would still leave the "why can't I put a corpse in the back seat" problem, unless the stored volume of cargo were to be subtracted from passenger capacity or vice versa.

**Comment by PatrikLundell** [comment]
You can't fit a corpse into the back seat for the same reason that you can't fit a lot of stuff into a flatbed truck, i.e. that each tile is its own storage. You'd need to do a lot of work to strap a corpse up such that it can "sit" in a back seat without encroaching on other seats. Do you really want to try to model how different shapes and combinations of shapes would fit into different spaces?
I'm perfectly fine with seats being essentially useless for item transport (although I'm not particularly happy with the military cargo truck not actually having any cargo space, but that's a separate issue).

Dealing with over/under size characters and mutants is better done with special cases for dealing with those, and that would probably involve creating new parts for such characters. Very small characters might be able to get disability controls from existing vehicles (for use by people that can't use their legs for control (and, I would presume, real world dwarfs), and so use hand controls for everything), while oversize characters likely would have to custom make their seats (possibly by introduce a back seat "seat extender" component for the back seat that marks the space as off limits for other seat stuff (and storage, of course), to indicate the seat is pushed back into the back seat's place). Prior to that I'd suggest just requiring there to be no roof (and later an extra high roof component, preferably with a drag penalty).

**Comment by zachary-kaelan** [comment]
The interactions between mutations and various other character parameters in general are all over the place.

* Some mutations increase strength. That increases body volume, max HP, and body mass, and by extension metabolic rate.
* Some mutations changes size category. That changes height, which changes BMI, metabolic rate, and volume.
* Some mutations directly change HP.
* Some mutations directly change metabolism.
* There's the `FAT` mutation which gives slight bash protection, warmth bonuses, and movement penalties.

I also don't believe there's currently an indicator on the character creation screen that height and age are more than just flavor.

**Comment by PatrikLundell** [comment]
No, there's not anything in the creation screen indicating size and age changing anything. However, weight affects things in the game, with the riding of ponies requires a total weight (with carried stuff) of at most 100 kg, and boats having a max carry weight. Both of these cases make real world sense (although the ponies being called "horses" is misleading).

**Comment by Maddremor** [comment]
> > I think the old model of aisles not providing any storage while seats provide a minimal amount is better than trying to use storage capacity for both storage capacity and seating capacity.
> 
> This would re-introducd old issues such as not being able to store things in seats which should reasonably fit there and oversized/overloaded characters being able to fit places they ought not to.
> 
> Splitting passenger capacity out to its own value would still leave the "why can't I put a corpse in the back seat" problem, unless the stored volume of cargo were to be subtracted from passenger capacity or vice versa.

An inability to fit a corpse in a seat isn't as urgent given that the player never controls a corpse, nor frequently transport them in such a fashion. The issues we are facing now are far more substantial.

**Comment by kevingranade** [comment]
Considering this doesn't address the issue for obese characters that *should* have no trouble fitting, and addressing that would push these values to even higher values, I think just deactivating the feature that prevents entry into these spaces as in #74981 is the right short-term fix, and before re enabling that we need to overhaul what values we're looking at and how those are represented in vehicle parts.
